src/pagination/PaginationButton.ts

```diff
@@ -53,42 +55,22 @@ export const usePaginationButton = createHook<
     };
   },
 
-  useProps(
-    { currentPage, move, last, first, next, prev, goto, getAriaLabel },
-    { onClick: htmlOnClick, ...htmlProps },
-  ) {
+  useProps(options, { onClick: htmlOnClick, ...htmlProps }) {
```

Comment:
I think we should not just destructure it like so..

Because this way options would also contain all other functions and if users passes the same name to the "goto" prop the function would execute, although it's not a huge issue, since users would get typescript error if they pass anything else.

```js
options[goto]?.();
```

Let's do `{ currentPage, move, goto, getAriaLabel, ...options },` since this way the options would only contain the "next", "prev", "last", "first" methods.


Or else we can also double check the function names if you want to be safe

```js

  if (isNumber(goto)) {
    move?.(goto);
  } else if (["next", "prev", "last", "first"].includes(goto)) {
    options[goto]?.();
  }
```

Reply:
I will go with the last one, much safer.